diff --git ./tables-3.1.1/tables/utilsextension.pyx~ ./tables-3.1.1/tables/utilsextension.pyx
index ebc0446d5f..01e5b3b790 100644
--- tables-3.1.1/tables/utilsextension.pyx~
+++ tables-3.1.1/tables/utilsextension.pyx
@@ -234,11 +234,6 @@ cdef register_blosc_():
 # This function detects if blosc can work correctly on the current platform.
 # This function has been written by Julian Taylor <jtaylor@ubuntu.com>.
 def _arch_without_blosc():
-    import platform
-    arch = platform.machine().lower()
-    for a in ["arm", "sparc", "mips"]:
-        if a in arch:
-            return True
     return False
 
 # Only register bloc compressor on platforms that actually support it.
